src/scicloj/metamorph/core.clj

```diff
@@ -1,17 +1,18 @@
 (ns scicloj.metamorph.core
   (:require [scicloj.metamorph.protocols :as prot]))
 
+(defn ^:deprecated uuid
+  "DEPRECATED: Use clojure.core/random-uuid instead"
+  []
+  (random-uuid))
 
-(defn uuid []
-  (java.util.UUID/randomUUID))
-
-
-(defn check-metamorph-compliant [ctx op]
+(defn check-metamorph-compliant
+  [ctx op]
   (cond
     (keyword? op) ctx
-    (not  (map? ctx)) (throw (IllegalArgumentException.  (str  "Metamorph pipe functions need to return a map, but returned: " ctx "of class: " (type ctx))))
-    (not  (contains? ctx :metamorph/data)) (do (println "Context after operation " op " with meta " (meta op) "does not contain :metamorph/data. This is likely as mistake.") ctx)
-    true ctx))
+    (not (map? ctx)) (throw (IllegalArgumentException.  (str  "Metamorph pipe functions need to return a map, but returned: " ctx "of class: " (type ctx))))
+    (not (contains? ctx :metamorph/data)) (do (println "Context after operation " op " with meta " (meta op) "does not contain :metamorph/data. This is likely as mistake.") ctx)
+    :else ctx))
 
 (defn pipeline
```

Comment:
Just asking before committing, I think that we can improve this function by dividing it into three smaller functions like this:

```suggestion
(defn- process-operation
  "Process an operation on the given current context.
  
  :param ctx: The current context that is being processed
  :param [id op]: A vector of operation ID and operation function
  :return: The updated context after processing the operation
  :throws IllegalArgumentException: If the operation is not a map or function"
  [ctx [id op]]
  (assert (some? op) "op cannot be nil")
  (cond
    (map? op) (merge ctx op)
    (ifn? op) (-> ctx
                  (assoc :metamorph/id (get ctx :metamorph/id id))
                  (op)
                  (check-metamorph-compliant op)
                  (dissoc :metamorph/id))
    :else (throw (IllegalArgumentException. (str "Cannot call a non function: " op)))))

(defn- local-pipeline
  "Execute operations on the given context.
  
  :param ops-with-id: A list of operations with their respective IDs
  :param ctx: The context to be processed by the operations
  :return: The updated context after executing all operations"
  ([ops-with-id] (local-pipeline ops-with-id {}))
  ([ops-with-id ctx]
   (let [ctx (if-not (map? ctx) {:metamorph/data ctx} ctx)]
     (reduce process-operation ctx ops-with-id))))

(defn pipeline
  "Create a metamorph pipeline function out of a list of operations.
  
  :param ops: A list of metamorph compliant operations (functions taking a context as first argument)
  :return: A function that can be executed with a context to process the operations"
  [& ops]
  (->> ops
       (mapv #(vector (uuid) %))
       (partial local-pipeline)))
```

Let me know what do you think about it to add it into the PR if it makes sense (or not) @behrica

Reply:
We have detailed tests, so if they still pass.
But maybe better in a different PR, so we can rollback easier, if needed.

Reply:
I did not write the code myself.
So I can not judge, if it breaks anything.

We have detailed tests, so if they still pass.
But maybe better in a different PR, so we can rollback easier, if needed.